pandas/tests/scalar/timestamp/test_unary_ops.py

```diff
@@ -118,6 +118,27 @@ def test_ceil_floor_edge(self, test_input, rounder, freq, expected):
             expected = Timestamp(expected)
             assert result == expected
 
+    @pytest.mark.parametrize('test_input, freq, expected', [
+        ('2018-01-01 00:02:06', '2s', '2018-01-01 00:02:06'),
+        ('2018-01-01 00:02:00', '2T', '2018-01-01 00:02:00'),
+        ('2018-01-01 00:04:00', '4T', '2018-01-01 00:04:00'),
+        ('2018-01-01 00:15:00', '15T', '2018-01-01 00:15:00'),
+        ('2018-01-01 00:20:00', '20T', '2018-01-01 00:20:00'),
+        ('2018-01-01 03:00:00', '3H', '2018-01-01 03:00:00'),
+    ])
+    def test_round_minute_freq(self, test_input, freq, expected):
+        # Ensure timestamps that shouldnt round dont!
+        # GH#21262
+        dt = Timestamp(test_input)
+        expected = Timestamp(expected)
+
+        result_ceil = dt.ceil(freq)
```

Comment:
Could you also parametrize over the rounding methods (`ceil`, `floor`, and `round`)? It would help reduce this duplication

Reply:
Sure makes sense to me